packet/routes/api.py

```diff
@@ -1,19 +1,31 @@
-from packet import auth, app
-from packet.utils import before_request
-from packet.packet import sign as sign_packet
+from packet import app, db
+from packet.utils import before_request, packet_auth
+from packet.models import Packet, MiscSignature
 
 
-@app.route("/api/v1/<member_username>/sign/<packet_username>", methods=["POST"])
-@auth.oidc_auth
+@app.route("/api/v1/sign/<packet_id>/", methods=["POST"])
+@packet_auth
 @before_request
-def sign(member_username, packet_username, info):
-    if info['uid'] != member_username:
-        if info['member_info']:
-            if "eboard-evaluations" not in info['member_info']['group_list']:
-                return "Error: You are not evals"
+def sign(packet_id, info):
+    packet = Packet.by_id(packet_id)
+
+    if packet is not None and packet.is_open():
+        if app.config["REALM"] == "csh":
+            # Check if the CSHer is an upperclassman and if so, sign that row
+            for sig in filter(lambda sig: sig.member == info["uid"], packet.upper_signatures):
+                sig.signed = True
+                db.session.commit()
+                return "Success: Signed Packet: " + packet.freshman_username
```

Comment:
This should return a parsable JSON response with an explicit `200 OK` status code, e.g.

```
HTTP/1.1 200 OK
Content-Type: application/json

{
    "status": "success",
    "message": "Successfully signed [username]'s packet"
}
```

Ultimately, what this should really do is return the updated array of signatures to update the DOM without a page refresh (and preferably coupled with optimistic UI updates).

Reply:
One of the future changes is to switch over to a REST API. However, I didn't want to do that in this PR as it's already quite big. As a results, I decided to leave the output as-is for now.